### PR TITLE
Feature/feature clear app

### PR DESCRIPTION
**Pull request by ArtMaZa**

### CODE REVIEW

File: src/components/App.jsx [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+      tasks: [],
+      id: 0,
+      value:''
+    };
+  }
+
+  removeListItem = (id) => {
+   let newTasks = [];
+   for (const key in this.state.tasks) {
+     if (!(id === this.state.tasks[key].id)) {
```
Comment (mhryshkin) [inline src/components/App.jsx:26]: please use array built-in functions

File: src/components/App.jsx [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+   }
+   this.setState(() => ({
+     tasks: newTasks,
+   }));  
+  }
+
+  add = () => {
+    if (this.state.value) {
+      this.setState((prevState) => ({
+        id: prevState.id + 1,
```
Comment (mhryshkin) [inline src/components/App.jsx:38]: why do you need id here?

File: src/components/App.jsx [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+import List from './List.jsx';
+import '../styles/App.css';
+import '../styles/style.css';
+
+class App extends Component {
+  static propsTypes = {
+    tasks: PropTypes.array,
+    id: PropTypes.number,
+    value: PropTypes.string
+  }
```
Comment (mhryshkin) [inline src/components/App.jsx:12]: Does this component have 'tasks, id, value' props?

File: src/components/ListItem.jsx [hunk 4]
```diff
@@ -0,0 +1,18 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+function ListItem({task, removeListItem}) {
+  const remove = (e) => removeListItem(task.id, e);
```
Comment (mhryshkin) [inline src/components/ListItem.jsx:5]: handleRemove

**Review by mhryshkin — CHANGES_REQUESTED** [review]

File: src/components/ListItem.jsx [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+  return <li className="list__item">
+    <input className="list__input" value={task.value} readOnly />
+    <button className="list__remove" onClick={remove}> X </button>
+  </li>;
+}
+
+ListItem.propTypes = {
+  task: PropTypes.object,
+  removeListItem: PropTypes.func,
```
Comment (mhryshkin) [inline src/components/ListItem.jsx:14]: use 'required' options for propTypes if a value is required

File: src/components/App.jsx (hunk 3 above)
Comment (ArtMaZa) [reply src/components/App.jsx:21]: well, I need to give each field a unique id number that will be used in the keys and in the component id

File: src/components/App.jsx (hunk 2 above)
Comment (ArtMaZa) [reply src/components/App.jsx:67]: well, I need to give each field a unique id number that will be used in the keys and in the component id

**Pull request merged**